backend/webapi/Models/Party.cs

```diff
@@ -51,6 +51,27 @@ public class Party : BaseAuditable
     [Projectable]
     public string FullName => $"{this.FirstName} {this.LastName}";
 
+    /// <summary>
+    /// First name to display.
+    /// The preferred first name if provided otherwise the first name.
+    /// </summary>
+    [Projectable]
+    public string DisplayFirstName => !string.IsNullOrWhiteSpace(this.PreferredFirstName) ? this.PreferredFirstName : this.FirstName;
+
+    /// <summary>
+    /// Last name to display.
+    /// The preferred last name if provided otherwise the last name.
+    /// </summary>
+    [Projectable]
+    public string DisplayLastName => !string.IsNullOrWhiteSpace(this.PreferredLastName) ? this.PreferredLastName : this.LastName;
```

Comment:
Using `IsNullOrWhitespace` here works but makes the resulting SQL much more complicated unfortunately. if we instead did `this.PreferredLastName ?? this.LastName;` It changes the SQL for preferred full name from

```SQL
SELECT CASE
          WHEN ((p."PreferredFirstName" IS NOT NULL)) AND (btrim(p."PreferredFirstName", E' \t\n\r') <> '') THEN p."PreferredFirstName"
          ELSE p."FirstName"
      END, CASE
          WHEN ((p."PreferredLastName" IS NOT NULL)) AND (btrim(p."PreferredLastName", E' \t\n\r') <> '') THEN p."PreferredLastName"
          ELSE p."LastName"
      END
```
to
```SQL
SELECT COALESCE(p."PreferredFirstName", p."FirstName"), COALESCE(p."PreferredLastName", p."LastName")
```

But also, it kinda muddies the waters about whether a white-space preferred name is acceptable as a value vs. whether it is acceptable bussiness-logic wise.

Currently in this PR the frontend allows submitting white-space names
But the backend converts that into a null value
And then still checks for a white-space name here

If this was instead just `this.PreferredLastName ?? this.LastName;` it now shows white-space preferred names (if that's what the business wants); and if that isn't what we want we add validation to prevent white-space names.